### ISSUE TITLE
Options related to should_replace and prefered

### ISSUE THREAD

**Issue by Noktam** [pymedusa/Medusa#5805]
TLDR :
Hi, i want to suggest tree options for the config search :

"When searching for prefered quality, allow lower allowed quality to be replace by better allowed quality" at yes by default.
(It's a simple revert back option from this : https://github.com/pymedusa/Medusa/issues/4511 )

and this two :

"When searching for better quality, always replace release who don't contains preferred words by one who has it, even if it's lower quality and don't ever replace a rlz with preferred with one without even if better quality"

"Don't stop backlog until you get a release with prefered words"

------------------------------------------------

Hi. I'm french, so i watch most of the time tvshow with french subtitles.
Unlike english subtitles, french one tend to be a lot more often out of sync (or even unaivailiable)

I could put VOSTFR and SUBFRENCH as required words, but it doesn't really suit my need, i have to wait en extra beetween 1 and 7 days to watch an episode.

The desired behavior for my usecase is : download wathever 720p show who come first, get the sub (english and or french) as quick as you can. And when you get a VOSTFR files, remplace the english one. Even if it's lower quality (like replace a Webdl with en HDTV with VOSTFR tag).

I guess i'm not the only one who would be interested by this case.

Right now i'm doing that really quick and dirty hack :

On common.py under def name_quality(name, anime=False, extend=True):

        if name.lower().find(".vostfr.") > -1 and name.lower().find(".720p.") > -1:
            return Quality.HDBLURAY
        if name.lower().find(".subfrench.") > -1 and name.lower().find(".720p.") > -1:
            return Quality.HDBLURAY
        if name.lower().find(".multi.") > -1 and name.lower().find(".720p.") > -1:
            return Quality.HDBLURAY
        if name.lower().find(".vostfr.") > -1 and name.lower().find(".1080p.") > -1:
            return Quality.FULLHDBLURAY
        if name.lower().find(".subfrench.") > -1 and name.lower().find(".1080p.") > -1:
            return Quality.FULLHDBLURAY
        if name.lower().find(".multi.") > -1 and name.lower().find(".1080p.") > -1:
            return Quality.FULLHDBLURAY

On parser.py under :     def get_quality(self, guess, extend=False):

        if self.original_name.lower().find(".vostfr.") > -1 or self.original_name.lower().find(".subfrench.") > -1 or self.original_name.lower().find(".multi.") > -1:
            return common.Quality.name_quality(self.original_name, self.is_anime, extend)

But i'm kinda annoyed by a change made on medusa here :
https://github.com/pymedusa/Medusa/issues/4511

I usually go like this, on allowed i have : 720P HDTV, 720P WEBDL, 720P Blueray and 1080P Blueray and prefered : 720P blueray

I want that medusa grab any of the allowed quality and change only if a prefered is availiable. I really don't care about replacing an HDTV by a WEBDL.

So i think the original behavior should be availiable at least as an option for people who don't want to trump files between allowed quality, only to prefered.

Like : 
"When searching for prefered quality, allow lower allowed quality to be replace by better allowed quality" at yes by default.

Now, to get back on my VOSTFR issue, it's a long time since i didn't update my sickrage (because, you know, i have to edit it every time), so i'm new on medusa (and fuck yeah, it's awesome, thanks for the manual search).
I see there is now a "preferred words", it's nice but not good enought.

I would suggest two extra options :

"When searching for better quality, always replace release who don't contains preferred words by one who has it, even if it's lower quality and don't ever replace a rlz with preferred with one without even if better quality"

"Don't stop backlog until you get a release with prefered words"

**Comment by medariox** [comment]
All already requested.
